### PR TITLE
SA-587 Adding support for backwards compatability checks

### PR DESCRIPTION
**Pull request by atomfinger**
## Description

Adding GitHub Actions job to verify OpenAPI backwards compatibility.

Do note that the backwards compatibility job will fail in this PR as there's no previous OpenAPI file to compare.

### CODE REVIEW

**Comment by atomfinger** [comment]
Current dependencies on/for this PR:
* main
  * **PR #167** <a href="https://app.graphite.dev/github/pr/protectorinsurance/protector-initializr-java/167" target="_blank">Graphite</a> 
    * **PR #168** <a href="https://app.graphite.dev/github/pr/protectorinsurance/protector-initializr-java/168" target="_blank">Graphite</a> 
      * **PR #169** <a href="https://app.graphite.dev/github/pr/protectorinsurance/protector-initializr-java/169" target="_blank">Graphite</a> 
        * **PR #170** <a href="https://app.graphite.dev/github/pr/protectorinsurance/protector-initializr-java/170" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/protectorinsurance/protector-initializr-java/170?utm_source=stack-comment).

File: .github/workflows/openapi-compatibility-check.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          name: head-openapi.yaml
+      - uses: actions/download-artifact@v3
+        id: download-base
+        with:
+          name: base-openapi.yaml
+      - name: Run OpenAPI Diff (from HEAD rev)
+        uses: docker://openapitools/openapi-diff:latest
+        with:
+          args: --fail-on-incompatible base-openapi.yaml head-openapi.yaml
+# INITIALIZR:WEB
```
Comment (pfs-olwe99) [inline .github/workflows/openapi-compatibility-check.yaml:75]: Avsluta med newline

Comment (atomfinger) [reply .github/workflows/openapi-compatibility-check.yaml:75]: Skal fikse, men er det at den ikke har newline nok til å ikke godkjene eller gi response på PR?

https://protectorinsurance.atlassian.net/wiki/spaces/IDP/pages/2111897823/Code+review+guidelines#Never-leave-a-review-without-a-resolution

File: .github/workflows/openapi-compatibility-check.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'gradle'
+      - name: Generate head schema
+        run: ./gradlew :web-schema:build
+      - name: Rename schema file
+        run: mv ./web-schema/build/generated/openapi/openapi.yaml head-openapi.yaml
+      - uses: actions/upload-artifact@v3
+        name: Uploading head schema artefact
```
Comment (pfs-olwe99) [inline .github/workflows/openapi-compatibility-check.yaml:28]: Typo: artefact -> artifact

File: .github/workflows/openapi-compatibility-check.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'gradle'
+      - name: Generate head schema
+        run: ./gradlew :web-schema:build
+      - name: Rename schema file
+        run: mv ./web-schema/build/generated/openapi/openapi.yaml base-openapi.yaml
+      - uses: actions/upload-artifact@v3
+        name: Uploading head schema artefact
```
Comment (pfs-olwe99) [inline .github/workflows/openapi-compatibility-check.yaml:53]: Typo: artefact -> artifact

File: .github/workflows/openapi-compatibility-check.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        with:
+          name: base-openapi.yaml
+          path: base-openapi.yaml
+
+  detect-breaking-changes:
+    name: Validate backwards compatability
```
Comment (pfs-olwe99) [inline .github/workflows/openapi-compatibility-check.yaml:59]: Typo: compatability-> compatibility

**Review by pfs-olwe99 — APPROVED** [review]

File: .github/workflows/openapi-compatibility-check.yaml (hunk 3 above)
Comment (atomfinger) [reply .github/workflows/openapi-compatibility-check.yaml:53]: > Artefact is the original British English spelling. Artifact is the American English spelling

I tend to use the former, but I'll fix it.

**Comment by atomfinger** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/protectorinsurance/protector-initializr-java/170?utm_source=stack-comment).


Job Reference: g1ehmV5GBeycWsnRsZGe

**Review by krgu99 — APPROVED** [review]

**Comment by atomfinger** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/protectorinsurance/protector-initializr-java/170?utm_source=stack-comment).


Job Reference: E8fU2eQsyKOI1omDiSkT

**Pull request merged**